### PR TITLE
PB-2077 - GeoIP Functionality (Cloudflare Supported)

### PR DESCRIPTION
**Pull request by wrcroberts**
### Goals
- **[EP-251](https://globalpersonals.atlassian.net/browse/EP-251)** General Platform Improvements
- **[PB-2077](https://globalpersonals.atlassian.net/browse/PB-2077):** Use CloudFlare (GEOIP) for non-Design Tool landing pages

Implementing a GeoIP script supported by Cloudflare to be used on landing pages to ensure that users sign up/logging in to the correct site.

Compatible with SSL and non-SSL landing pages.

***Progress: 90%***

---

### Tasks

- [x] Two configuration files: one (geoip.config.php) which allows customisation with your own WLD
site ids and urls.
- [x] Example index file as a placeholder to show what code needs to go into a landing page file.
- [x] https://github.com/globaldev/php-cloudflare-geoip/pull/2 Merge refactored code (to satisfy older version of PHP).
- [x] Two example code files for SSL and non-SSL implementation.
- [x] Instructions document for implementation.
- [x] Update README.md.

### CODE REVIEW

File: cloudflare-geoip/geoip.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Geoip Script (utlizing Cloudflare variable)
```
Comment (simonbingham) [inline cloudflare-geoip/geoip.php]: Typo "utlizing".

Comment (wrcroberts) [reply cloudflare-geoip/geoip.php]: 🙈 Addressing - https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/37461c20b6f9ee8ba1b7275a45a8bba9837f59f8

**Comment by wrcroberts** [comment]
@timtait After our discussion Friday:

- Removed all the country toggles and replaced with ability to uncomment countries to enable them https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/405e68152f797e80279995a6a6a2e732b88b6e4d#diff-6e16ff2a7b4bedb836a4149ade7d045bL2

- Removed country settings within geoip.php file itself and replaced with a check that the constant is defined instead https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/405e68152f797e80279995a6a6a2e732b88b6e4d#diff-8181715c758c6260471e6da6d42c13e0L17.

File: index.example.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+require 'cloudflare-geoip/geoip.php';
+$geoip = new Geoip();
+$country = (isset($_SERVER["HTTP_CF_IPCOUNTRY"]) ? $_SERVER["HTTP_CF_IPCOUNTRY"] : "GB");
```
Comment (timtait) [inline index.example.php]: Should that fallback be `GB` or `UK` as we've got a mix now. 

We're using `UK` - https://github.com/globaldev/php-cloudflare-geoip/pull/1/files#diff-8181715c758c6260471e6da6d42c13e0R18 
but we are using `GB` here - https://github.com/globaldev/php-cloudflare-geoip/pull/1/files#diff-6e16ff2a7b4bedb836a4149ade7d045bR14

Comment (wrcroberts) [reply index.example.php]: 👋 977c70e Made this change which should address the issue (may need to be tidied in terms of the array).

Only reason I changed to 'GB' and 'ZA' is due to the country codes that Cloudflare uses.

Good point though as it breaks the `set_vars` function. It should now use the Cloudflare version of country codes to do the comparison and then use UK, SA, etc to assign as the `WLD_COUNTRY_CODE`

File: cloudflare-geoip/geoip.config.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    # 3. Enter the WLD Site URL (excluding http:// or https://) for your countries in place of premium.whitelabeldating.com.
+
+    $country_site["AU"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # Australia
+    $country_site["CA"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # Canada
+    $country_site["IE"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # Ireland
+    $country_site["NZ"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # New Zealand
+    $country_site["ZA"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # South Africa
+    $country_site["GB"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # United Kingdom
+    $country_site["US"] = array("WLD_SITE_ID"=>"0000", "WLD_SITE_URL"=>"premium.whitelabeldating.com"); # United States
+
```
Comment (gpearce) [inline cloudflare-geoip/geoip.config.php]: Can we use `app.example.com` or similar as the example.

File: index.example.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
```
Comment (gpearce) [inline index.example.php]: Don't we / can't we use the const `DEFAULT_COUNTRY` here? Or remove this altogether and rely on the GeoIP class to define a default country based on the config file?

Comment (gpearce) [reply index.example.php]: We should also add a 'fuller' example IMHO that shows the various options being chucked into a basic registration/login flow.

File: cloudflare-geoip/geoip.config.php (hunk 3 above)
Comment (wrcroberts) [reply cloudflare-geoip/geoip.config.php]: Should be addressed here - https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/a3e52df534292e3e64a012d385b7a8f8aba0d47d

File: index.example.php (hunk 4 above)
Comment (wrcroberts) [reply index.example.php]: First point should be addressed here - https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/2cea4de25b4cb83f8359e41b4c3f0fd9f7dc7a9e.
Now checking that `DEFAULT_COUNTRY` is defined and assigning it correctly and then using that variable to perform the Cloudflare check before going onto the variable assignment for the WLD SITE ID and URLs.

Comment (wrcroberts) [reply index.example.php]: Second point - https://github.com/globaldev/php-cloudflare-geoip/pull/1/commits/19345eed79eb7c6456c5b283ce8421074c957acb.
Includes two new files that show the implementation set up of SSL and non-SSL landing page components.
May need to update the documentation to reflect this 👀

**Pull request merged**